controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -295,6 +308,66 @@ public void getCurrentSegments(StreamInfo request, StreamObserver<SegmentRanges>
                 responseObserver);
     }
 
+    private Supplier<String> createDelegationTokenSupplier(StreamInfo request) {
+        return () -> {
+            StreamAuthParams authParams = new StreamAuthParams(request.getScope(), request.getStream(),
+                    this.isInternalWritesWithReadPermEnabled);
+
+            // StreamResource will take on values containing `_RG` (ex: "prn:://scope:myScope/stream:_RGmyApp")
+            // and `_MARK` for internal streams. We use the same strings in delegation tokens sent to Segment Store,
+            // while we use resource tags lie "reader-group" (ex: "prn:://scope:myScope/reader-group:myApp)
+            // and "watermark" for authorization in Controller.
+            String streamResource = authParams.streamResourceString();
+
+            // This is for backward compatibility.
+            if (request.getRequestedPermission().equals("") && authParams.isStreamUserDefined()) {
```

Comment:
I suppose it should be empty, since its a new String based field in the protobuf schema which should take an empty value when the server parses the message coming from an older client that doesn't populate it: 

```
message StreamInfo {
    string scope = 1;
    string stream = 2;
    string requestedPermission = 3;
}
```

According to https://developers.google.com/protocol-buffers/docs/proto3, 

> ...
> When a message is parsed, if the encoded message does not contain a particular singular element, the corresponding field in the parsed object is set to the default value for that field. These defaults are type-specific:
>
> For strings, the default value is the empty string.
> ...

Does that make sense?